main.go

```diff
@@ -123,11 +125,15 @@ func init() {
 }
 
 func main() {
+	os.Exit(innerMain())
```

Comment:
is this refactor just so we don't call `os.Exit()` in a bunch of places?

Reply:
Correct, `os.Exit()` breaks `defer`, so `defer close()` would not reliably be called without this.